### PR TITLE
Removed all the deprecated python training code and related tests and utils

### PR DESCRIPTION
**Pull request by askhade**
### Description
Motivation for this PR is code cleanup.

1. Remove all deprecated python code related to orttrainer, old checkpoint, related tests and utils
2. Cleanup orttraining_pybind_state.cc to remove all deprecated bindings.

### CODE REVIEW

**Review by er3x3 — COMMENTED** [review]
Looks like below two files can also be removed:
- onnxruntime/test/python/onnxruntime_test_training_unittest_utils.py
- orttraining/orttraining/python/checkpointing_utils.py

**Comment by xadupre** [comment]
It looks good to me. Should we remove this code in two steps: first a warning then a true deletion?

File: orttraining/orttraining/test/python/orttraining_ortmodule_tests.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def run_data_sampler_tests(cwd, log):
 def run_hooks_tests(cwd, log):
     log.debug("Running: Data hooks tests")
 
-    command = [sys.executable, "-m", "pytest", "-sv", "orttraining_test_hooks.py"]
+    command = [sys.executable, "-m", "pytest", "-sv", "orttraining_test_ortmodule_hooks.py"]
```
Comment (pengwa) [inline orttraining/orttraining/test/python/orttraining_ortmodule_tests.py:149]: nit: strictly speaking, the hooks here is training/utils/hooks, not bound to ortmodule.

**Review by pengwa — APPROVED** [review]
LGTM. We should add a note in new release for the deprecation information.

**Comment by askhade** [comment]
> It looks good to me. Should we remove this code in two steps: first a warning then a true deletion?

The warning has been in place for quite some time now: https://github.com/microsoft/onnxruntime/pull/13022/files. This is the second step.

**Review by pengwa — APPROVED** [review]

**Pull request merged**